examples/docs_snippets/docs_snippets_tests/guides_tests/reexecution_tests/test_unreliable_pipeline.py

```diff
@@ -0,0 +1,7 @@
+from dagster import execute_pipeline
+from docs_snippets.guides.dagster.reexecution.pipeline.unreliable_pipeline import unreliable_pipeline
+
+
+def test_pipeline_compiles_and_executes():
+    result = execute_pipeline(unreliable_pipeline)
+    assert result
```

Comment:
we will need `assert result.success` instead.
result will always be not null because it returns an execution result including several metadata -  when the execution fails, it'd return an execution result whose `success` attribute is false.

Reply:
Had a question here - the pipeline is designed to fail 50% of the time. When the execution fails, will `result.success` by false?

Reply:
hmm that's a good point. Yes, when the execution fails, `result.success` will be false. Can we make it so the execution result can be deterministic or in the test? Just throwing ideas:
- we could pass in a bool solid config and set the default to be "will fail", and in tests, we can configure the bool value so the solid will not fail, e.g.
```
@solid(config_schema={"will_fail": Field(bool, is_required=False, default_value=True)})
def unreliable(context, num):
    failure_rate = 0.5
    if context.solid_config['will_fail'] and random() < failure_rate:
        raise Exception("blah")
```

- alternatively we can only execution the subset which won't fail in this test via `execute_pipeline(unreliable_pipeline, solid_selection=[...])`

Your call - whichever you feel would work the best with the user-facing content.

Reply:
Makes sense, I updated the test to just run the two solids in the pipeline that should actually succeed all the time.